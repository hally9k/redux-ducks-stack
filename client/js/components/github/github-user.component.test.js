import GithubUser from 'components/github/github-user.component';
import React from 'react';
import { mount, shallow } from 'enzyme';

const props = {
    user: {
        avatar: 'https://myimage.com',
        username: 'hally9k',
        id: '123456'
    }
};

describe('components', () => {
    describe('GithubSearch', () => {
        it('should render the user when given a valid user object', () => {
            const enzymeWrapper = mount(<GithubUser {...props} />);

            expect(enzymeWrapper.find('div.github-user img').prop('src')).toBe('https://myimage.com');
            expect(enzymeWrapper.find('div.github-user h2').text()).toBe('hally9k');
            expect(enzymeWrapper.find('div.github-user h4').text()).toBe('ID: 123456');
        });

        it('mounts the component', () => {
            const enzymeWrapper = shallow(<GithubUser {...props} />);
            expect(enzymeWrapper).toMatchSnapshot();
        });
    });
});
