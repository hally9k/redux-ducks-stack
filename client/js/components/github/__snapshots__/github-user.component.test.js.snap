// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components GithubSearch mounts the component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="github-user"
>
    <img
        src="https://myimage.com"
    />
    <h2>
        hally9k
    </h2>
    <h4>
        ID: 
        123456
    </h4>
</div>,
  "nodes": Array [
    <div
      className="github-user"
>
      <img
            src="https://myimage.com"
      />
      <h2>
            hally9k
      </h2>
      <h4>
            ID: 
            123456
      </h4>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <GithubUser
        user={
                Object {
                        "avatar": "https://myimage.com",
                        "id": "123456",
                        "username": "hally9k",
                      }
        }
/>,
      "_debugID": 9,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": GithubUser {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "user": Object {
            "avatar": "https://myimage.com",
            "id": "123456",
            "username": "hally9k",
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 4,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="github-user"
>
          <img
                    src="https://myimage.com"
          />
          <h2>
                    hally9k
          </h2>
          <h4>
                    ID: 
                    123456
          </h4>
</div>,
        "_debugID": 10,
        "_renderedOutput": <div
          className="github-user"
>
          <img
                    src="https://myimage.com"
          />
          <h2>
                    hally9k
          </h2>
          <h4>
                    ID: 
                    123456
          </h4>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <GithubUser
    user={
        Object {
            "avatar": "https://myimage.com",
            "id": "123456",
            "username": "hally9k",
          }
    }
/>,
}
`;
